lib/exceptions.rb

```diff
@@ -170,6 +170,7 @@ class UpgradeFailed < MachineryError; end
     class BinaryDiffError < MachineryError; end
 
     class SshConnectionFailed < MachineryError; end
+    class SshConnectionDisrupted < SystemExit; end
```

Comment:
Also, is it really a problem if the inspection continues? The problem could have been temporary and the next inspection works just fine, no?